src/FrozenClock.php

```diff
@@ -22,6 +22,11 @@ public function setTo(DateTimeImmutable $now): void
         $this->now = $now;
     }
 
+    public function modify(string $modify): void
```

Comment:
I know you're trying to keep the name aligned with the underlying object, but we're operating in a different level (clock) and this might raise questions like: modify? modify what?
`FrozenClock#setTo()` was chosen as a way to replicate what we do with clocks ("I'll set the clock to X").

Can you think of a clearer name for this operation?
Also, please add a doc block to clarify the expectations around the value passed there, like:

```suggestion
    /**
     * @param non-empty-string $modify
     * @throws DateMalformedStringException When an invalid date/time string is passed (PHP 8.3+)
     *
     * @see https://www.php.net/manual/en/datetime.formats.php
     */
    public function modify(string $modify): void
```